plugins/cassandra/cassandraembed.cpp

```diff
@@ -2566,6 +2660,138 @@ class AttributeMapColumnMapper : implements CassandraColumnMapper
     }
 } attributeMapColumnMapper;
 
+class ElementMapColumnMapper : implements CassandraColumnMapper
+{
+public:
+    virtual IPTree *toXML(IPTree *row, const char *name, const CassValue *value)
+    {
+        CassandraIterator elems(cass_iterator_from_map(value));
+        while (cass_iterator_next(elems))
+        {
+            rtlDataAttr str;
+            unsigned chars;
+            getStringResult(NULL, cass_iterator_get_map_key(elems), chars, str.refstr());
+            StringBuffer elemName(chars, str.getstr());
```

Comment:
If this is time critical then elemName could be declared outside, and set - otherwise clearer as it is.